engines/sci/engine/script_patches.cpp

```diff
@@ -7356,11 +7356,42 @@ static const uint16 qg4SetLooperPatch2[] = {
 	PATCH_END
 };
 
+// The panel showing time of day gets stuck displaying consecutive moonrises.
+// Despite time actually advancing, the sun will never rise again because
+// local5 is set to 1 for night UI and never resets until hero moves to
+// another room.
+//
```

Comment:
Can you please add the explanation you wrote in the bug - you wrote that temp 0 is not used in this function, thus we can safely ignore and reuse the code block that manipulates it in order to fix this bug.

Without this explanation, it’s not clear why we are removing and reusing this code block altogether